File: services/user-service/src/main/java/com/codeconnect/user/application/dto/response/MentorApprovalResponse.java
package com.codeconnect.user.application.dto.response;

import com.codeconnect.user.domain.enums.MentorApprovalStatus;

import java.time.Instant;

/**
 * Immutable response projection for mentor onboarding applications.
 */
public record MentorApprovalResponse(
    String id,
    String userId,
    String email,
    String linkedInUrl,
    String bio,
    MentorApprovalStatus status,
    Instant submittedAt,
    Instant reviewedAt,
    String reviewedBy
) {}

File: frontend/src/dto/response/MentorApprovalResponse.ts
import { MentorApprovalStatus } from "@/domain/enums/MentorApprovalStatus";

/**
 * Representation of pending mentor application record matching:
 * com.codeconnect.user.application.dto.MentorApprovalResponse
 */
export interface MentorApprovalResponse {
  readonly id: string;
  readonly userId: string;
  readonly email: string;
  readonly linkedInUrl: string | null;
  readonly bio: string | null;
  readonly status: MentorApprovalStatus;
  readonly submittedAt: string;
  readonly reviewedAt: string | null;
  readonly reviewedBy: string | null;
}
